src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -499,6 +507,17 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                                 {
                                     if (isRestoreSucceeded)
                                     {
+                                        bool shouldDisplayVulnerabilitiesInfoBar = LogUtility.AreVulnerabilitiesInRestoreSummaries(restoreSummaries);
+
+                                        if (shouldDisplayVulnerabilitiesInfoBar)
+                                        {
+                                            await _infoBarService.Value.Show(t);
+                                        }
+                                        else if (_infoBarService.IsValueCreated) // if the InfoBar was created and all vulnerabilities are fixed, hide it.
```

Comment:
>and all vulnerabilities are fixed [](http://example.com/codeflow?start=111&length=33)

Suggestion from what you explained: ...but all vulnerabilities were fixed as we were displaying the InfoBar, hide it.